addons/account/models/account_payment.py

```diff
@@ -702,6 +705,100 @@ def _compute_payment_receipt_title(self):
+               FROM account_move_line
+               JOIN account_payment
+                 ON account_payment.move_id = account_move_line.move_id) AS move_line
+        """)
+        suspense_account_id = self.company_id.account_journal_suspense_account_id.id
+        outstanding_account_id = self.outstanding_account_id.id
+
+        if not self[0].id:  # if record is under creation/edition in UI, safely inject values in the query
+            # Necessary since new record aren't searchable in the DB and record in edition aren't up to date yet
+            place_holders = {
+                'move_id': self._origin.move_id.id or 0,
+                'payment_id': self._origin.id or 0,
+                'payment_type': self.payment_type,
+                'balance': self.amount if self.payment_type == 'outbound' else -self.amount,
+                'account_id': self.destination_account_id.id,
+                'outstanding_account_id': outstanding_account_id,
+                'company_id': self.company_id.id or None,
+                'date': self.date or None,
+                'partner_id': self.partner_id.id,
+            }
+            # In case of null values, postgres may have issues with the column type, so we cast the columns
+            move_table_and_alias = SQL("""
+                (VALUES (%(move_id)s::int4, %(payment_id)s::int4, %(payment_type)s::varchar, %(balance)s::numeric, %(account_id)s::int4, %(company_id)s::int4, %(date)s::date, %(partner_id)s::int4),
+                        (%(move_id)s::int4, %(payment_id)s::int4, %(payment_type)s::varchar, -1.0 * %(balance)s::numeric, %(outstanding_account_id)s::int4, %(company_id)s::int4, %(date)s::date, %(partner_id)s::int4)
+                ) AS move_line(move_id, payment_id, payment_type, balance, account_id, company_id, date, partner_id)
+            """, **place_holders)
+
+        query = SQL(
+            """
+            SELECT
+                   move_line.payment_id,
+                   array_agg(DISTINCT dup_move_line.move_id) AS duplicate_move_ids
+              FROM %(move_table_and_alias)s
+              JOIN account_move_line AS dup_move_line
+                ON move_line.move_id != dup_move_line.move_id
+               AND move_line.partner_id = dup_move_line.partner_id
+               AND move_line.company_id = dup_move_line.company_id
+               AND move_line.date = dup_move_line.date
+               AND dup_move_line.parent_state IN %(matching_states)s
+               AND (move_line.account_id = dup_move_line.account_id OR dup_move_line.account_id = %(suspense_account_id)s)
+               AND NOT dup_move_line.reconciled
+             WHERE move_line.payment_id IN %(payments)s
+               AND (
+                   -- Case 1: a move is a credit in same account receivable or debit in same acc payable as the payment
+                   (move_line.account_id != %(outstanding_account_id)s AND move_line.balance = dup_move_line.balance)
```

Comment:
`outstanding_account_id` seems like it can be empty, and so False

Reply:
@qdp-odoo it was failing in a new test pushed after my last changes. Now, instead of checking self.outstanding_account_id, check self.company_id.account_journal_payment_debit_account_id 
 and self.company_id.account_journal_payment_credit_account